modules/bluefish/bluefish.cpp

```diff
@@ -78,21 +75,42 @@ std::vector<std::wstring> device_list()
 
 void init(core::module_dependencies dependencies)
 {
+
+	bool bCanBeConsumer = false;
 	try
 	{
-		blue_initialize();
+		BvcWrapper blue = BvcWrapper();
+		int numCards = 0;
+		blue.Enumerate(numCards);
+		if (numCards)
+		{
+			for (int n = 1; n < (numCards + 1); n++)
+			{
+				unsigned int value;
+				blue.Attach(n);
+				blue.QueryCardProperty32(CARD_FEATURE_STREAM_INFO, value);
+				if (CARD_FEATURE_GET_SDI_OUTPUT_STREAM_COUNT(value))
+					bCanBeConsumer = true;
+
+				blue.Detach();
+			}
+		}	
 	}
 	catch(...){}
 
-	dependencies.consumer_registry->register_consumer_factory(L"Bluefish Consumer", create_consumer, describe_consumer);
-	dependencies.consumer_registry->register_preconfigured_consumer_factory(L"bluefish", create_preconfigured_consumer);
-	dependencies.system_info_provider_repo->register_system_info_provider([](boost::property_tree::wptree& info)
+	if (bCanBeConsumer)
 	{
-		info.add(L"system.bluefish.version", version());
-
-		for (auto device : device_list())
-			info.add(L"system.bluefish.device", device);
-	});
+		dependencies.consumer_registry->register_consumer_factory(L"Bluefish Consumer", create_consumer, describe_consumer);
+		dependencies.consumer_registry->register_preconfigured_consumer_factory(L"bluefish", create_preconfigured_consumer);
+		dependencies.system_info_provider_repo->register_system_info_provider([](boost::property_tree::wptree& info)
+		{
+			info.add(L"system.bluefish.version", version());
```

Comment:
This is also an example of something that should be included even though Bluefish drivers are not installed on a system so that the startup log will show `bluefish.version Not found`